### PR TITLE
Feat/cardano graphql services setup

### PR DESCRIPTION
**Pull request by rhyslbw**
This PR isolates the initial setup of this package, which is currently sourced in #197, implementing the following feedback:

- https://github.com/input-output-hk/cardano-js-sdk/pull/197#pullrequestreview-905409263
- https://github.com/input-output-hk/cardano-js-sdk/pull/197#discussion_r823414376

### CODE REVIEW

**Comment by jvieiro** [comment]
LGTM

File: packages/cardano-graphql-services/src/util/logging.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * Appends an object containing module metadata to each log entry
+ *
+ */
+export const moduleLogger = (logger: Logger, module: string): Logger => {
+  const methodNames = ['debug', 'error', 'info', 'trace', 'warn'] as (keyof Logger)[];
+  return <Logger>(<unknown>Object.fromEntries(
+    methodNames.map((method) => [
+      method,
+      (...optionalParams: any[]) => {
```
Comment (mkazlauskas) [inline packages/cardano-graphql-services/src/util/logging.ts:12]: Lint warning here

File: packages/cardano-graphql-services/test/util/stubLogger.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Logger } from 'ts-log';
+
+// eslint-disable-next-line unicorn/consistent-function-scoping,@typescript-eslint/no-empty-function
+const noOp = () => {};
+
+export const createStubLogger = (): Logger => ({
```
Comment (mkazlauskas) [inline packages/cardano-graphql-services/test/util/stubLogger.ts:6]: Might be a good idea to hoist this to `util-dev`

File: packages/cardano-graphql-services/test/util/stubLogger.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  debug: noOp,
```
Comment (mkazlauskas) [inline packages/cardano-graphql-services/test/util/stubLogger.ts:7]: Is there a reason to not just use `jest.fn()`? Wouldn't need to `spyOn` then.

**Review by mkazlauskas — CHANGES_REQUESTED** [review]

File: packages/cardano-graphql-services/src/util/logging.ts (hunk 1 above)
Comment (rhyslbw) [reply packages/cardano-graphql-services/src/util/logging.ts:13]: Resolved https://github.com/input-output-hk/cardano-js-sdk/pull/202/commits/d93a121c626e7c9ce060d575802bc2775cf875e3#diff-db57d698466df51e35405ca4cb76a9d20944a81ed30c1056d43fe1bcc57e1504R12

File: packages/cardano-graphql-services/test/util/stubLogger.ts (hunk 2 above)
Comment (rhyslbw) [reply packages/cardano-graphql-services/test/util/stubLogger.ts]: https://github.com/input-output-hk/cardano-js-sdk/pull/202/commits/d93a121c626e7c9ce060d575802bc2775cf875e3#diff-190bd16bc1fbf12703a9d5509d897ac1c69b5e8e39dae8a2fe7914820343e22bR3-R9

File: packages/cardano-graphql-services/test/util/stubLogger.ts (hunk 3 above)
Comment (rhyslbw) [reply packages/cardano-graphql-services/test/util/stubLogger.ts]: https://github.com/input-output-hk/cardano-js-sdk/pull/202/commits/d93a121c626e7c9ce060d575802bc2775cf875e3#diff-190bd16bc1fbf12703a9d5509d897ac1c69b5e8e39dae8a2fe7914820343e22bR3-R9

**Pull request merged**